js/ui/windowManager.js

```diff
@@ -310,6 +311,149 @@ var EOSShellWobbly = new Lang.Class({
+    Extends: GObject.Object,
+    Properties: {
+        length: GObject.ParamSpec.uint('length',
+                                       'Length',
+                                       'How long the animation lasts',
+                                       GObject.ParamFlags.READWRITE |
+                                       GObject.ParamFlags.CONSTRUCT,
+                                       0,
+                                       10000,
+                                       200)
+    },
+
+    vfunc_get_name: function() {
+        return 'zoom';
+    },
+
+    createActorPrivate: function(actor) {
+        let effect = new AttachedZoomAnimation({ bridge: this, enabled: false });
+        actor.add_effect_with_name('endless-animation-zoom', effect);
+        return effect;
+    }
+});
+
+// AttachedZoomAnimation
+//
+// A zoom animation as attached to an actor.
+const AttachedZoomAnimation = new Lang.Class({
+    Name: 'AttachedZoomAnimation',
+    Extends: EndlessShellFX.Affine,
+    Implements: [ AnimationsDbus.ServerSurfaceAttachedEffect ],
+    Properties: {
+        'bridge': GObject.ParamSpec.object('bridge',
+                                           'Zoom Animation Bridge',
+                                           'The bridge to the settings',
+                                           GObject.ParamFlags.READWRITE |
+                                           GObject.ParamFlags.CONSTRUCT_ONLY,
+                                           ControllableZoomAnimation)
+    },
+
+    _init: function(params) {
+        this.parent(params);
+        this._enabledId = -1;
+    },
+
+    _startAnimation: function(animation, done) {
+        this.transform_animation = animation;
+        this.set_enabled(true);
+
+        this._enabledId = this.connect('notify::enabled', () => {
+            if (!this.enabled) {
+                done();
+                this.disconnect(this._enabledId);
+                this.enabledId = -1;
+            }
+        });
+    },
+
+    activate: function(event, detail) {
+        let [x, y] = this.actor.get_position();
+        let [width, height] = this.actor.get_size();
+
+        switch (event) {
+            case 'open':
+                this._startAnimation(new Animation.ZoomAnimation({
+                    to: new Animation.Box({
+                        top_left: new Animation.Vector({ x: x, y: y }),
+                        bottom_right: new Animation.Vector({ x: x + width, y: y + height })
+                    }),
+                    from: new Animation.Box({
+                        top_left: new Animation.Vector({ x: detail.pointer[0], y: detail.pointer[1] }),
+                        bottom_right: new Animation.Vector({ x: detail.pointer[0], y: detail.pointer[1] })
+                    }),
+                    stepper: new Animation.LinearStepper({ length: this.bridge.length })
+                }), detail.complete);
+                return true;
+                break;
```

Comment:
If you have `return` you don't need a `break` here. Also, this switch statement seems a bit confusing. Would it look better as a regular `if , else if …` set of blocks?

Reply:
Hmm okay. I think someone else suggested `return` and `break` in the same block. I think it'd be better to have the switch statement - we're picking from a range of events here.

Reply:
Fixed